llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -1357,6 +1357,13 @@ bool SITargetLowering::isLegalGlobalAddressingMode(const AddrMode &AM) const {
   return isLegalMUBUFAddressingMode(AM);
 }
 
+bool SITargetLowering::isLegalFlatScratchAddressingMode(
```

Comment:
As an alternative, you could change `isLegalFlatAddressingMode` to take `AS` and `Variant` as arguments.

Reply:
Done in the latest commit.